### PR TITLE
Program Test: clean up invoke_builtin_function

### PR DESCRIPTION
**Pull request by andreisilviudragnea**
#### Problem
`invoke_builtin_function` had some duplicated code that could be removed.

#### Summary of Changes
I cleaned up `invoke_builtin_function`.

Originally a part of #2635.

### CODE REVIEW

**Review by buffalojoec — DISMISSED** [review]
Looks good, thanks!

File: program-test/src/lib.rs [hunk 1]
```diff
@@ -156,7 +153,6 @@ pub fn invoke_builtin_function(
 
     // Re-fetch the instruction context. The previous reference may have been
     // invalidated due to the `set_invoke_context` in a CPI.
-    let transaction_context = &invoke_context.transaction_context;
```
Comment (joncinque) [inline program-test/src/lib.rs]: We can't remove this line. `set_invoke_context` overwrites `invoke_context` with unsafe code, so there's a possibility of invalidating the previous reference to `transaction_context`. The rest is good though!

Comment (joncinque) [reply program-test/src/lib.rs]: https://github.com/solana-labs/solana/pull/26507 added this logic, but unfortunately figuring out the issue is lost in a slack chat

**Review by joncinque — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**